From: Michael Froman <mfroman@mozilla.com>
Date: Mon, 19 Jan 2026 13:45:21 -0600
Subject: (tmp-cherry-pick) Revert "Privatize sigslots in NetworkManager"
 (fbd0bcbe2d)

This reverts commit 12168786cc61f58792acaa02ac917aa1ea996801.

Reason for revert: Missed a reference in Chromium

Original change's description:
> Privatize sigslots in NetworkManager
>
> To be submitted after Chromium adaptations (now landed):
> https://chromium-review.googlesource.com/c/chromium/src/+/7172038
>
> Bug: webrtc:42222066
> Change-Id: Ic3f0d2c717bc38dc13db4e1b8cd6e788bfc1df7c
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/426980
> Reviewed-by: Tomas Gunnarsson <tommi@webrtc.org>
> Commit-Queue: Harald Alvestrand <hta@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#46262}

Bug: webrtc:42222066
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Change-Id: I7c04eaaafa6a403ba1253cec72e18c904c64e564
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/427061
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Commit-Queue: Tomas Gunnarsson <tommi@webrtc.org>
Reviewed-by: Tomas Gunnarsson <tommi@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#46272}
---
 rtc_base/network.h | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/rtc_base/network.h b/rtc_base/network.h
index 442dca5928..57ac29f800 100644
--- a/rtc_base/network.h
+++ b/rtc_base/network.h
@@ -187,6 +187,9 @@ class RTC_EXPORT NetworkManager : public DefaultLocalAddressProvider,
 
   virtual void set_vpn_list(const std::vector<NetworkMask>& /* vpn */) {}
 
+  sigslot::signal0<> SignalNetworksChanged;
+  sigslot::signal0<> SignalError;
+
   // The implementation of the Subscribe methods is in the .cc file due
   // to linking issues with Chrome.
   void SubscribeNetworksChanged(absl::AnyInvocable<void()> callback);
@@ -199,9 +202,6 @@ class RTC_EXPORT NetworkManager : public DefaultLocalAddressProvider,
   void NotifyError() { SignalError(); }
 
  private:
-  sigslot::signal0<> SignalNetworksChanged;
-  sigslot::signal0<> SignalError;
-
   SignalTrampoline<NetworkManager, &NetworkManager::SignalNetworksChanged>
       networks_changed_trampoline_;
   SignalTrampoline<NetworkManager, &NetworkManager::SignalError>
@@ -525,7 +525,8 @@ class RTC_EXPORT NetworkManagerBase : public NetworkManager {
 // Basic implementation of the NetworkManager interface that gets list
 // of networks using OS APIs.
 class RTC_EXPORT BasicNetworkManager : public NetworkManagerBase,
-                                       public NetworkBinderInterface {
+                                       public NetworkBinderInterface,
+                                       public sigslot::has_slots<> {
  public:
   BasicNetworkManager(
       const Environment& env,
